tensorflow_datasets/image/fruits360.py

```diff
@@ -0,0 +1,111 @@
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import re
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """\
+@article{article,
+author = {Mureșan, Horea and Oltean, Mihai},
+year = {2018},
+month = {06},
+pages = {26-42},
+title = {Fruit recognition from images using deep learning},
+volume = {10},
+journal = {Acta Universitatis Sapientiae, Informatica},
+doi = {10.2478/ausi-2018-0002},
+url = {https://github.com/Horea94/Fruit-Images-Dataset}
+}
+"""
+
+_DOWNLOAD_URL = "https://github.com/Horea94/Fruit-Images-Dataset/archive/8c839e4d52ae0563d9e7d3986f34113e52a910a5.zip"
+
+_IMAGE_SIZE = 100
+_IMAGE_SHAPE = (_IMAGE_SIZE, _IMAGE_SIZE, 3)
+_CLASS_NAMES = ['Apple Braeburn', 'Apple Golden 1', 'Apple Golden 2', 'Apple Golden 3', 'Apple Granny Smith', 'Apple Red 1', 'Apple Red 2', 'Apple Red 3', 'Apple Red Delicious', 'Apple Red Yellow 1', 'Apple Red Yellow 2', 'Apricot', 'Avocado', 'Avocado ripe', 'Banana', 'Banana Lady Finger', 'Banana Red', 'Cactus fruit', 'Cantaloupe 1', 'Cantaloupe 2', 'Carambula', 'Cherry 1', 'Cherry 2', 'Cherry Rainier', 'Cherry Wax Black', 'Cherry Wax Red', 'Cherry Wax Yellow', 'Chestnut', 'Clementine', 'Cocos', 'Dates', 'Granadilla', 'Grape Blue', 'Grape Pink', 'Grape White', 'Grape White 2', 'Grape White 3', 'Grape White 4', 'Grapefruit Pink', 'Grapefruit White', 'Guava', 'Hazelnut', 'Huckleberry', 'Kaki', 'Kiwi', 'Kumquats', 'Lemon', 'Lemon Meyer', 'Limes', 'Lychee', 'Mandarine', 'Mango', 'Mangostan', 'Maracuja', 'Melon Piel de Sapo', 'Mulberry', 'Nectarine', 'Orange', 'Papaya', 'Passion Fruit', 'Peach', 'Peach 2', 'Peach Flat', 'Pear', 'Pear Abate', 'Pear Kaiser', 'Pear Monster', 'Pear Williams', 'Pepino', 'Physalis', 'Physalis with Husk', 'Pineapple', 'Pineapple Mini', 'Pitahaya Red', 'Plum', 'Plum 2', 'Plum 3', 'Pomegranate', 'Pomelo Sweetie', 'Quince', 'Rambutan', 'Raspberry', 'Redcurrant', 'Salak', 'Strawberry', 'Strawberry Wedge', 'Tamarillo', 'Tangelo', 'Tomato 1', 'Tomato 2', 'Tomato 3', 'Tomato 4', 'Tomato Cherry Red', 'Tomato Maroon', 'Walnut']
+
+
+class Fruits360(tfds.core.GeneratorBasedBuilder):
+  """Fruits-360 dataset."""
+
+  VERSION = tfds.core.Version("0.0.2")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description="A large set of fruits on a white background.",
+        features=tfds.features.FeaturesDict({
+            "file_name": tfds.features.Text(),  # Eg: 'n15075141_54.JPEG'
+            "label": tfds.features.ClassLabel(
+                names=_CLASS_NAMES),
+        }),
+        supervised_keys=("image", "label"),
+        urls=["https://github.com/Horea94/Fruit-Images-Dataset/"],
+        citation=_CITATION
+        )
+
+  def _split_generators(self, dl_manager):
+    # For some reason the default extract method is NO_EXTRACT
+    # so we specify it here. :/
+    download_resource = tfds.download.Resource(
```

Comment:
are you sure this is necessary? it should auto-detect the extraction method based on the file name (.zip). try removing.

Reply:
According to my tests, this is necessary. I'm not sure why but extraction failed without it earlier. I'll try again in a moment.